klaviyo/public.py

```diff
@@ -63,8 +72,31 @@ def track(
         if ip_address:
             params['ip'] = ip_address
 
-        query_string = self._build_query_string(params, is_test)
-        return self._public_request(self.TRACK, query_string)
+        if method not in ['get','post']:
+
+            raise ValueError('method argument must be either "post" or "get"')
+
+        else:
+
+            if method == 'post':
+
+                url = '{}/{}'.format(KlaviyoAPI.KLAVIYO_API_SERVER,self.TRACK)
+
+                headers = {
+                    "Accept": "text/html",
+                    "Content-Type": "application/x-www-form-urlencoded"
+                }
+
+                transformed = 'data='+quote(dumps(params))
+
+                response = request("POST", url, data=transformed, headers=headers)
+
+                return KlaviyoAPIResponse(response.status_code, response.json())
```

Comment:
the logic here is duplicated down in the identify method. Could you dry this out a bit? perhaps a public_post_request or something similar available in the api class?

Reply:
created a separate _build_query_string_post and _public_post_request functions in api_helper